src/components/Personalization/in-app-message-actions/actions/displayIframeContent.js

```diff
@@ -148,60 +145,75 @@ export const createIframe = (htmlContent, clickHandler) => {
   return element;
 };
 
-export const createContainerElement = settings => {
-  const { mobileParameters = {}, webParameters = {} } = settings;
-  const element = document.createElement("div");
-  element.id = ELEMENT_TAG_ID;
-  element.className = `${ELEMENT_TAG_CLASSNAME}`;
-  Object.assign(
-    element.style,
-    buildStyleFromParameters(mobileParameters, webParameters)
-  );
-
-  return element;
-};
-
-export const createOverlayElement = parameter => {
-  const element = document.createElement("div");
-  const backdropOpacity = parameter.backdropOpacity || 0.5;
-  const backdropColor = parameter.backdropColor || "#FFFFFF";
-  element.id = OVERLAY_TAG_ID;
-  element.className = `${OVERLAY_TAG_CLASSNAME}`;
-
-  Object.assign(element.style, {
-    position: "fixed",
-    top: "0",
-    left: "0",
-    width: "100%",
-    height: "100%",
-    background: "transparent",
-    opacity: backdropOpacity,
-    backgroundColor: backdropColor
-  });
-
-  return element;
-};
-
 export const displayHTMLContentInIframe = (settings, interact) => {
   dismissMessage();
-  const { content, contentType, mobileParameters } = settings;
+  const { content, contentType, mobileParameters, webParameters } = settings;
 
   if (contentType !== TEXT_HTML) {
     return;
   }
 
-  const container = createContainerElement(settings);
+  const container = createElement(ALLOY_MESSAGING_CONTAINER_ID);
 
   const iframe = createIframe(content, createIframeClickHandler(interact));
 
-  container.appendChild(iframe);
+  const overlay = createElement(ALLOY_OVERLAY_CONTAINER_ID);
 
-  if (mobileParameters.uiTakeover) {
-    const overlay = createOverlayElement(mobileParameters);
-    document.body.appendChild(overlay);
-    document.body.style.overflow = "hidden";
+  container.appendChild(iframe);
+  if (webParameters && webParameters.info !== "this is a placeholder") {
+    Object.assign(iframe.style, webParameters[ALLOY_IFRAME_ID].style);
+    Object.assign(
+      container.style,
+      webParameters[ALLOY_MESSAGING_CONTAINER_ID].style
+    );
+    if (webParameters[ALLOY_OVERLAY_CONTAINER_ID].params.enabled) {
+      Object.assign(
+        overlay.style,
+        webParameters[ALLOY_OVERLAY_CONTAINER_ID].style
+      );
+      document.body.appendChild(overlay);
+      document.body.style.overflow = "hidden";
+    }
+    const parentElementSelector =
+      webParameters[ALLOY_MESSAGING_CONTAINER_ID].params.parentElement;
```

Comment:
I suggest using object destructuring for the `style` and `params`. objects.  FOr example...

```js
const {style={}, params={}} = webParameters[ALLOY_MESSAGING_CONTAINER_ID];
```
☝️ this guarantees that style and params always exist.  And if you create some new functions can use them as named.

Reply:
done